### DISCUSSION TITLE
tree-sitter for Haskell

### DISCUSSION THREAD

**Discussion by p3rsik** [helix-editor/helix#2920] [Q&A]
Hello! I was looking through the list of the supported languages and found out that tree-sitter isn't enabled(working?) for Haskell. May I ask why is that?

**Comment by the-mikedavis** [comment]
tree-sitter-haskell was disabled for a while because it caused build issues: https://github.com/helix-editor/helix/commit/59f94d13b8d61ce338ecf4957c6b4ad945021009.

It was re-enabled in https://github.com/helix-editor/helix/pull/1417 which was released in `22.03`. If you're running version `v0.6.0` or older, haskell syntax highlighting is disabled.

Haskell syntax highlighting is enabled and working since that PR, though, and from a quick test seems to work normally on master or 22.05 (the current latest release).

**Reply by p3rsik** [reply]
Hey!

Yeah, syntax highlighting is working, I'm asking about features like "expand selection to parent node". Or did I miss something and that is not a tree-sitter feature?

**Reply by the-mikedavis** [reply]
Yep, that's a tree-sitter feature that should work out of the box for Haskell. If that doesn't work consistently you may be seeing the behavior fixed in https://github.com/helix-editor/helix/pull/2877

Some tree-sitter driven features like function/class/comment textobjects need extra queries which are not written yet (needs a `runtime/queries/haskell/textobjects.scm` file with queries).

**Reply by p3rsik** [reply]
Thanks! Will wait for the version with that, I've got my answers)

Also thanks for the amazing editor :)

**Reply by p3rsik** [reply]
Wait, actually, where can I found the documentation on how to write a `textobjects.scm` file for Haskell?

**Reply by p3rsik** [reply]
Okay, I found [it](https://docs.helix-editor.com/master/guides/textobject.html).
